File: specs/master/query_dsl/geo/distance/geo_distance_query.ts

/**namespace:QueryDsl.Geo.Distance */
/**custom_serialization*/
interface GeoDistanceQuery {
	Location: GeoLocation;
	distance: Distance;
	optimize_bbox: GeoOptimizeBBox;
	distance_type: GeoDistanceType;
	coerce: boolean;
	ignore_malformed: boolean;
	validation_method: GeoValidationMethod;
}

File: output/csharp/QueryDsl/Geo/Distance/GeoDistanceQuery.cs
using Nest.CommonOptions;
using Nest.QueryDsl;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.QueryDsl {

	public class GeoDistanceQuery  {
		
		[DataMember(Name="distance")]
		public Distance Distance { get; set; }


		[DataMember(Name="distance_type")]
		public GeoDistanceType DistanceType { get; set; }


		[DataMember(Name="location")]
		public GeoLocation Location { get; set; }


		[DataMember(Name="validation_method")]
		public GeoValidationMethod ValidationMethod { get; set; }

	}
}
